{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "cf7ba11c",
   "metadata": {},
   "source": [
    "Due to libary conflicts this notebook is executed using an different environment. \n",
    "```\n",
    "conda install -c bioconda anndata2ri\n",
    "conda install nb_conda_kernels\n",
    "conda install -c bioconda scanpy \n",
    "```"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "470fbbca",
   "metadata": {},
   "outputs": [],
   "source": [
    "import scanpy as sc\n",
    "import anndata as ann\n",
    "import pandas as pd\n",
    "import numpy as np"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "c88a4c4a",
   "metadata": {},
   "outputs": [],
   "source": [
    "import anndata2ri\n",
    "anndata2ri.activate()\n",
    "%load_ext rpy2.ipython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "1bbfdf4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_raw_base = '../data/Borcherding/raw/'\n",
    "path_out_base = '../data/Borcherding/'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2951d5b9",
   "metadata": {},
   "source": [
    "## Convert rds to h5ad"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "b7adddde",
   "metadata": {},
   "outputs": [],
   "source": [
    "path_combinedtcr = path_raw_base + 'CombinedTCR_object.rds'  #\n",
    "path_filtered = path_raw_base + 'filtered_seuratObjects_harmony-001.rds'  #\n",
    "path_gse123 = path_raw_base + 'GSE123814.rds' #\n",
    "path_gse144 = path_raw_base + 'GSE144469_TCR_filtered_contig_annotations_all.csv'  # "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "37d69837",
   "metadata": {},
   "outputs": [],
   "source": [
    "out_combined = path_out_base + 'tcrs_combined.csv'\n",
    "out_filtered = path_out_base + 'rna_combined.h5ad'\n",
    "out_gse123 = path_out_base + 'rna_123.h5ad'\n",
    "out_gse144 = path_out_base + 'tcrs_144.csv'"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "8ab9e195",
   "metadata": {},
   "source": [
    "### TCR Data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "20cb765b",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i path_combinedtcr -o combined_tcr\n",
    "combined_tcr <- readRDS(path_combinedtcr)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "936908bb",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>sample</th>\n",
       "      <th>ID</th>\n",
       "      <th>TCR1</th>\n",
       "      <th>cdr3_aa1</th>\n",
       "      <th>cdr3_nt1</th>\n",
       "      <th>TCR2</th>\n",
       "      <th>cdr3_aa2</th>\n",
       "      <th>cdr3_nt2</th>\n",
       "      <th>CTgene</th>\n",
       "      <th>CTnt</th>\n",
       "      <th>CTaa</th>\n",
       "      <th>CTstrict</th>\n",
       "      <th>cellType</th>\n",
       "      <th>Patient</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>BCT1.1_AAACCTGCAGATCGGA-1</td>\n",
       "      <td>BCT1.1</td>\n",
       "      <td>ID</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>TRBV4-1..TRBJ2-5.TRBC2</td>\n",
       "      <td>CASRLAGGLQETQYF</td>\n",
       "      <td>TGCGCCAGCCGCCTAGCGGGGGGGCTCCAAGAGACCCAGTACTTC</td>\n",
       "      <td>NA_TRBV4-1..TRBJ2-5.TRBC2</td>\n",
       "      <td>NA_TGCGCCAGCCGCCTAGCGGGGGGGCTCCAAGAGACCCAGTACTTC</td>\n",
       "      <td>NA_CASRLAGGLQETQYF</td>\n",
       "      <td>NA_NA_TRBV4-1..TRBJ2-5.TRBC2_TGCGCCAGCCGCCTAGC...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>BCT1.1_AAACGGGTCATAGCAC-1</td>\n",
       "      <td>BCT1.1</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV13-2.TRAJ3.TRAC</td>\n",
       "      <td>CAETILYSSASKIIF</td>\n",
       "      <td>TGTGCAGAGACTATTTTGTACAGCAGTGCTTCCAAGATAATCTTT</td>\n",
       "      <td>TRBV30..TRBJ1-6.TRBC1</td>\n",
       "      <td>CAWTTPGTSNSPLHF</td>\n",
       "      <td>TGTGCCTGGACCACCCCCGGGACCAGTAATTCACCCCTCCACTTT</td>\n",
       "      <td>TRAV13-2.TRAJ3.TRAC_TRBV30..TRBJ1-6.TRBC1</td>\n",
       "      <td>TGTGCAGAGACTATTTTGTACAGCAGTGCTTCCAAGATAATCTTT_...</td>\n",
       "      <td>CAETILYSSASKIIF_CAWTTPGTSNSPLHF</td>\n",
       "      <td>TRAV13-2.TRAJ3.TRAC_TGTGCAGAGACTATTTTGTACAGCAG...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>BCT1.1_AAAGATGAGACAGGCT-1</td>\n",
       "      <td>BCT1.1</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV26-1.TRAJ23.TRAC</td>\n",
       "      <td>CIVSLSLVIYNQGGKLIF</td>\n",
       "      <td>TGCATCGTCAGTCTTTCTCTTGTGATTTATAACCAGGGAGGAAAGC...</td>\n",
       "      <td>TRBV28..TRBJ2-7.TRBC2</td>\n",
       "      <td>CASSSSWEGSPGEQYF</td>\n",
       "      <td>TGTGCCAGCAGCTCATCCTGGGAGGGGAGTCCGGGCGAGCAGTACTTC</td>\n",
       "      <td>TRAV26-1.TRAJ23.TRAC_TRBV28..TRBJ2-7.TRBC2</td>\n",
       "      <td>TGCATCGTCAGTCTTTCTCTTGTGATTTATAACCAGGGAGGAAAGC...</td>\n",
       "      <td>CIVSLSLVIYNQGGKLIF_CASSSSWEGSPGEQYF</td>\n",
       "      <td>TRAV26-1.TRAJ23.TRAC_TGCATCGTCAGTCTTTCTCTTGTGA...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>BCT1.1_AAAGATGCACAAGCCC-1</td>\n",
       "      <td>BCT1.1</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV2.TRAJ9.TRAC</td>\n",
       "      <td>CAVERNTGGFKTIF</td>\n",
       "      <td>TGTGCTGTGGAGCGAAATACTGGAGGCTTCAAAACTATCTTT</td>\n",
       "      <td>TRBV7-8..TRBJ1-5.TRBC1</td>\n",
       "      <td>CASSQLGNGNQPQHF</td>\n",
       "      <td>TGTGCCAGCAGCCAATTGGGGAATGGCAATCAGCCCCAGCATTTT</td>\n",
       "      <td>TRAV2.TRAJ9.TRAC_TRBV7-8..TRBJ1-5.TRBC1</td>\n",
       "      <td>TGTGCTGTGGAGCGAAATACTGGAGGCTTCAAAACTATCTTT_TGT...</td>\n",
       "      <td>CAVERNTGGFKTIF_CASSQLGNGNQPQHF</td>\n",
       "      <td>TRAV2.TRAJ9.TRAC_TGTGCTGTGGAGCGAAATACTGGAGGCTT...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>BCT1.1_AAAGATGTCTGAGTGT-1</td>\n",
       "      <td>BCT1.1</td>\n",
       "      <td>ID</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>TRBV28..TRBJ1-1.TRBC1</td>\n",
       "      <td>CASSLPGARVAFF</td>\n",
       "      <td>TGTGCCAGCAGTTTGCCGGGGGCCCGGGTGGCTTTCTTT</td>\n",
       "      <td>NA_TRBV28..TRBJ1-1.TRBC1</td>\n",
       "      <td>NA_TGTGCCAGCAGTTTGCCGGGGGCCCGGGTGGCTTTCTTT</td>\n",
       "      <td>NA_CASSLPGARVAFF</td>\n",
       "      <td>NA_NA_TRBV28..TRBJ1-1.TRBC1_TGTGCCAGCAGTTTGCCG...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient99</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213257</th>\n",
       "      <td>SCT4.2_TTTGTCAAGTGTTGAA-1</td>\n",
       "      <td>SCT4.2</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV8-3.TRAJ43.TRAC</td>\n",
       "      <td>CAVGNNDMRF</td>\n",
       "      <td>TGTGCTGTGGGGAACAATGACATGCGCTTT</td>\n",
       "      <td>TRBV5-1.TRBD1.TRBJ2-6.TRBC2</td>\n",
       "      <td>CASSLAQGAGANVLTF</td>\n",
       "      <td>TGCGCCAGCAGCTTGGCCCAGGGTGCTGGGGCCAACGTCCTGACTTTC</td>\n",
       "      <td>TRAV8-3.TRAJ43.TRAC_TRBV5-1.TRBD1.TRBJ2-6.TRBC2</td>\n",
       "      <td>TGTGCTGTGGGGAACAATGACATGCGCTTT_TGCGCCAGCAGCTTG...</td>\n",
       "      <td>CAVGNNDMRF_CASSLAQGAGANVLTF</td>\n",
       "      <td>TRAV8-3.TRAJ43.TRAC_TGTGCTGTGGGGAACAATGACATGCG...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213258</th>\n",
       "      <td>SCT4.2_TTTGTCACAAATACAG-1</td>\n",
       "      <td>SCT4.2</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV38-1.TRAJ44.TRAC</td>\n",
       "      <td>CAFMKQATASKLTF</td>\n",
       "      <td>TGTGCTTTCATGAAGCAGGCTACTGCCAGTAAACTCACCTTT</td>\n",
       "      <td>TRBV12-3..TRBJ1-5.TRBC1</td>\n",
       "      <td>CASSLWAGEGQPQHF</td>\n",
       "      <td>TGTGCCAGCAGTTTATGGGCAGGGGAAGGTCAGCCCCAGCATTTT</td>\n",
       "      <td>TRAV38-1.TRAJ44.TRAC_TRBV12-3..TRBJ1-5.TRBC1</td>\n",
       "      <td>TGTGCTTTCATGAAGCAGGCTACTGCCAGTAAACTCACCTTT_TGT...</td>\n",
       "      <td>CAFMKQATASKLTF_CASSLWAGEGQPQHF</td>\n",
       "      <td>TRAV38-1.TRAJ44.TRAC_TGTGCTTTCATGAAGCAGGCTACTG...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213259</th>\n",
       "      <td>SCT4.2_TTTGTCAGTCTAACGT-1</td>\n",
       "      <td>SCT4.2</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV17.TRAJ56.TRAC</td>\n",
       "      <td>CATVTTGANSKLTF</td>\n",
       "      <td>TGTGCTACGGTCACTACTGGAGCCAATAGTAAGCTGACATTT</td>\n",
       "      <td>TRBV5-1..TRBJ2-1.TRBC2</td>\n",
       "      <td>CASSLEAASSYNEQFF</td>\n",
       "      <td>TGCGCCAGCAGCTTGGAGGCGGCGAGCTCCTACAATGAGCAGTTCTTC</td>\n",
       "      <td>TRAV17.TRAJ56.TRAC_TRBV5-1..TRBJ2-1.TRBC2</td>\n",
       "      <td>TGTGCTACGGTCACTACTGGAGCCAATAGTAAGCTGACATTT_TGC...</td>\n",
       "      <td>CATVTTGANSKLTF_CASSLEAASSYNEQFF</td>\n",
       "      <td>TRAV17.TRAJ56.TRAC_TGTGCTACGGTCACTACTGGAGCCAAT...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213260</th>\n",
       "      <td>SCT4.2_TTTGTCAGTTCAGTAC-1</td>\n",
       "      <td>SCT4.2</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV17.TRAJ22.TRAC</td>\n",
       "      <td>CATAGSARQLTF</td>\n",
       "      <td>TGTGCTACGGCGGGTTCTGCAAGGCAACTGACCTTT</td>\n",
       "      <td>TRBV12-3..TRBJ2-3.TRBC2</td>\n",
       "      <td>CASSLSVSTDTQYF</td>\n",
       "      <td>TGTGCCAGCAGCCTTAGCGTTAGCACAGATACGCAGTATTTT</td>\n",
       "      <td>TRAV17.TRAJ22.TRAC_TRBV12-3..TRBJ2-3.TRBC2</td>\n",
       "      <td>TGTGCTACGGCGGGTTCTGCAAGGCAACTGACCTTT_TGTGCCAGC...</td>\n",
       "      <td>CATAGSARQLTF_CASSLSVSTDTQYF</td>\n",
       "      <td>TRAV17.TRAJ22.TRAC_TGTGCTACGGCGGGTTCTGCAAGGCAA...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient98</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1213261</th>\n",
       "      <td>SCT4.2_TTTGTCATCCTTGCCA-1</td>\n",
       "      <td>SCT4.2</td>\n",
       "      <td>ID</td>\n",
       "      <td>TRAV8-2.TRAJ8.TRAC</td>\n",
       "      <td>CVVSDYTGFQKLVF</td>\n",
       "      <td>TGTGTTGTGAGTGACTACACAGGCTTTCAGAAACTTGTATTT</td>\n",
       "      <td>TRBV4-1..TRBJ2-3.TRBC2</td>\n",
       "      <td>CASSQGGEALKYF</td>\n",
       "      <td>TGCGCCAGCAGCCAAGGGGGGGAGGCCCTAAAGTATTTT</td>\n",
       "      <td>TRAV8-2.TRAJ8.TRAC_TRBV4-1..TRBJ2-3.TRBC2</td>\n",
       "      <td>TGTGTTGTGAGTGACTACACAGGCTTTCAGAAACTTGTATTT_TGC...</td>\n",
       "      <td>CVVSDYTGFQKLVF_CASSQGGEALKYF</td>\n",
       "      <td>TRAV8-2.TRAJ8.TRAC_TGTGTTGTGAGTGACTACACAGGCTTT...</td>\n",
       "      <td>T-AB</td>\n",
       "      <td>Patient98</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1213262 rows × 15 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                           barcode  sample  ID                  TCR1  \\\n",
       "0        BCT1.1_AAACCTGCAGATCGGA-1  BCT1.1  ID                  None   \n",
       "1        BCT1.1_AAACGGGTCATAGCAC-1  BCT1.1  ID   TRAV13-2.TRAJ3.TRAC   \n",
       "2        BCT1.1_AAAGATGAGACAGGCT-1  BCT1.1  ID  TRAV26-1.TRAJ23.TRAC   \n",
       "3        BCT1.1_AAAGATGCACAAGCCC-1  BCT1.1  ID      TRAV2.TRAJ9.TRAC   \n",
       "4        BCT1.1_AAAGATGTCTGAGTGT-1  BCT1.1  ID                  None   \n",
       "...                            ...     ...  ..                   ...   \n",
       "1213257  SCT4.2_TTTGTCAAGTGTTGAA-1  SCT4.2  ID   TRAV8-3.TRAJ43.TRAC   \n",
       "1213258  SCT4.2_TTTGTCACAAATACAG-1  SCT4.2  ID  TRAV38-1.TRAJ44.TRAC   \n",
       "1213259  SCT4.2_TTTGTCAGTCTAACGT-1  SCT4.2  ID    TRAV17.TRAJ56.TRAC   \n",
       "1213260  SCT4.2_TTTGTCAGTTCAGTAC-1  SCT4.2  ID    TRAV17.TRAJ22.TRAC   \n",
       "1213261  SCT4.2_TTTGTCATCCTTGCCA-1  SCT4.2  ID    TRAV8-2.TRAJ8.TRAC   \n",
       "\n",
       "                   cdr3_aa1  \\\n",
       "0                      None   \n",
       "1           CAETILYSSASKIIF   \n",
       "2        CIVSLSLVIYNQGGKLIF   \n",
       "3            CAVERNTGGFKTIF   \n",
       "4                      None   \n",
       "...                     ...   \n",
       "1213257          CAVGNNDMRF   \n",
       "1213258      CAFMKQATASKLTF   \n",
       "1213259      CATVTTGANSKLTF   \n",
       "1213260        CATAGSARQLTF   \n",
       "1213261      CVVSDYTGFQKLVF   \n",
       "\n",
       "                                                  cdr3_nt1  \\\n",
       "0                                                     None   \n",
       "1            TGTGCAGAGACTATTTTGTACAGCAGTGCTTCCAAGATAATCTTT   \n",
       "2        TGCATCGTCAGTCTTTCTCTTGTGATTTATAACCAGGGAGGAAAGC...   \n",
       "3               TGTGCTGTGGAGCGAAATACTGGAGGCTTCAAAACTATCTTT   \n",
       "4                                                     None   \n",
       "...                                                    ...   \n",
       "1213257                     TGTGCTGTGGGGAACAATGACATGCGCTTT   \n",
       "1213258         TGTGCTTTCATGAAGCAGGCTACTGCCAGTAAACTCACCTTT   \n",
       "1213259         TGTGCTACGGTCACTACTGGAGCCAATAGTAAGCTGACATTT   \n",
       "1213260               TGTGCTACGGCGGGTTCTGCAAGGCAACTGACCTTT   \n",
       "1213261         TGTGTTGTGAGTGACTACACAGGCTTTCAGAAACTTGTATTT   \n",
       "\n",
       "                                TCR2          cdr3_aa2  \\\n",
       "0             TRBV4-1..TRBJ2-5.TRBC2   CASRLAGGLQETQYF   \n",
       "1              TRBV30..TRBJ1-6.TRBC1   CAWTTPGTSNSPLHF   \n",
       "2              TRBV28..TRBJ2-7.TRBC2  CASSSSWEGSPGEQYF   \n",
       "3             TRBV7-8..TRBJ1-5.TRBC1   CASSQLGNGNQPQHF   \n",
       "4              TRBV28..TRBJ1-1.TRBC1     CASSLPGARVAFF   \n",
       "...                              ...               ...   \n",
       "1213257  TRBV5-1.TRBD1.TRBJ2-6.TRBC2  CASSLAQGAGANVLTF   \n",
       "1213258      TRBV12-3..TRBJ1-5.TRBC1   CASSLWAGEGQPQHF   \n",
       "1213259       TRBV5-1..TRBJ2-1.TRBC2  CASSLEAASSYNEQFF   \n",
       "1213260      TRBV12-3..TRBJ2-3.TRBC2    CASSLSVSTDTQYF   \n",
       "1213261       TRBV4-1..TRBJ2-3.TRBC2     CASSQGGEALKYF   \n",
       "\n",
       "                                                 cdr3_nt2  \\\n",
       "0           TGCGCCAGCCGCCTAGCGGGGGGGCTCCAAGAGACCCAGTACTTC   \n",
       "1           TGTGCCTGGACCACCCCCGGGACCAGTAATTCACCCCTCCACTTT   \n",
       "2        TGTGCCAGCAGCTCATCCTGGGAGGGGAGTCCGGGCGAGCAGTACTTC   \n",
       "3           TGTGCCAGCAGCCAATTGGGGAATGGCAATCAGCCCCAGCATTTT   \n",
       "4                 TGTGCCAGCAGTTTGCCGGGGGCCCGGGTGGCTTTCTTT   \n",
       "...                                                   ...   \n",
       "1213257  TGCGCCAGCAGCTTGGCCCAGGGTGCTGGGGCCAACGTCCTGACTTTC   \n",
       "1213258     TGTGCCAGCAGTTTATGGGCAGGGGAAGGTCAGCCCCAGCATTTT   \n",
       "1213259  TGCGCCAGCAGCTTGGAGGCGGCGAGCTCCTACAATGAGCAGTTCTTC   \n",
       "1213260        TGTGCCAGCAGCCTTAGCGTTAGCACAGATACGCAGTATTTT   \n",
       "1213261           TGCGCCAGCAGCCAAGGGGGGGAGGCCCTAAAGTATTTT   \n",
       "\n",
       "                                                  CTgene  \\\n",
       "0                              NA_TRBV4-1..TRBJ2-5.TRBC2   \n",
       "1              TRAV13-2.TRAJ3.TRAC_TRBV30..TRBJ1-6.TRBC1   \n",
       "2             TRAV26-1.TRAJ23.TRAC_TRBV28..TRBJ2-7.TRBC2   \n",
       "3                TRAV2.TRAJ9.TRAC_TRBV7-8..TRBJ1-5.TRBC1   \n",
       "4                               NA_TRBV28..TRBJ1-1.TRBC1   \n",
       "...                                                  ...   \n",
       "1213257  TRAV8-3.TRAJ43.TRAC_TRBV5-1.TRBD1.TRBJ2-6.TRBC2   \n",
       "1213258     TRAV38-1.TRAJ44.TRAC_TRBV12-3..TRBJ1-5.TRBC1   \n",
       "1213259        TRAV17.TRAJ56.TRAC_TRBV5-1..TRBJ2-1.TRBC2   \n",
       "1213260       TRAV17.TRAJ22.TRAC_TRBV12-3..TRBJ2-3.TRBC2   \n",
       "1213261        TRAV8-2.TRAJ8.TRAC_TRBV4-1..TRBJ2-3.TRBC2   \n",
       "\n",
       "                                                      CTnt  \\\n",
       "0         NA_TGCGCCAGCCGCCTAGCGGGGGGGCTCCAAGAGACCCAGTACTTC   \n",
       "1        TGTGCAGAGACTATTTTGTACAGCAGTGCTTCCAAGATAATCTTT_...   \n",
       "2        TGCATCGTCAGTCTTTCTCTTGTGATTTATAACCAGGGAGGAAAGC...   \n",
       "3        TGTGCTGTGGAGCGAAATACTGGAGGCTTCAAAACTATCTTT_TGT...   \n",
       "4               NA_TGTGCCAGCAGTTTGCCGGGGGCCCGGGTGGCTTTCTTT   \n",
       "...                                                    ...   \n",
       "1213257  TGTGCTGTGGGGAACAATGACATGCGCTTT_TGCGCCAGCAGCTTG...   \n",
       "1213258  TGTGCTTTCATGAAGCAGGCTACTGCCAGTAAACTCACCTTT_TGT...   \n",
       "1213259  TGTGCTACGGTCACTACTGGAGCCAATAGTAAGCTGACATTT_TGC...   \n",
       "1213260  TGTGCTACGGCGGGTTCTGCAAGGCAACTGACCTTT_TGTGCCAGC...   \n",
       "1213261  TGTGTTGTGAGTGACTACACAGGCTTTCAGAAACTTGTATTT_TGC...   \n",
       "\n",
       "                                        CTaa  \\\n",
       "0                         NA_CASRLAGGLQETQYF   \n",
       "1            CAETILYSSASKIIF_CAWTTPGTSNSPLHF   \n",
       "2        CIVSLSLVIYNQGGKLIF_CASSSSWEGSPGEQYF   \n",
       "3             CAVERNTGGFKTIF_CASSQLGNGNQPQHF   \n",
       "4                           NA_CASSLPGARVAFF   \n",
       "...                                      ...   \n",
       "1213257          CAVGNNDMRF_CASSLAQGAGANVLTF   \n",
       "1213258       CAFMKQATASKLTF_CASSLWAGEGQPQHF   \n",
       "1213259      CATVTTGANSKLTF_CASSLEAASSYNEQFF   \n",
       "1213260          CATAGSARQLTF_CASSLSVSTDTQYF   \n",
       "1213261         CVVSDYTGFQKLVF_CASSQGGEALKYF   \n",
       "\n",
       "                                                  CTstrict cellType    Patient  \n",
       "0        NA_NA_TRBV4-1..TRBJ2-5.TRBC2_TGCGCCAGCCGCCTAGC...     T-AB  Patient99  \n",
       "1        TRAV13-2.TRAJ3.TRAC_TGTGCAGAGACTATTTTGTACAGCAG...     T-AB  Patient99  \n",
       "2        TRAV26-1.TRAJ23.TRAC_TGCATCGTCAGTCTTTCTCTTGTGA...     T-AB  Patient99  \n",
       "3        TRAV2.TRAJ9.TRAC_TGTGCTGTGGAGCGAAATACTGGAGGCTT...     T-AB  Patient99  \n",
       "4        NA_NA_TRBV28..TRBJ1-1.TRBC1_TGTGCCAGCAGTTTGCCG...     T-AB  Patient99  \n",
       "...                                                    ...      ...        ...  \n",
       "1213257  TRAV8-3.TRAJ43.TRAC_TGTGCTGTGGGGAACAATGACATGCG...     T-AB  Patient98  \n",
       "1213258  TRAV38-1.TRAJ44.TRAC_TGTGCTTTCATGAAGCAGGCTACTG...     T-AB  Patient98  \n",
       "1213259  TRAV17.TRAJ56.TRAC_TGTGCTACGGTCACTACTGGAGCCAAT...     T-AB  Patient98  \n",
       "1213260  TRAV17.TRAJ22.TRAC_TGTGCTACGGCGGGTTCTGCAAGGCAA...     T-AB  Patient98  \n",
       "1213261  TRAV8-2.TRAJ8.TRAC_TGTGTTGTGAGTGACTACACAGGCTTT...     T-AB  Patient98  \n",
       "\n",
       "[1213262 rows x 15 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "combined_tcr = list(combined_tcr)\n",
    "combined_tcr = pd.concat(combined_tcr, ignore_index=True)\n",
    "combined_tcr.to_csv(out_combined)\n",
    "combined_tcr"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1167b0b3",
   "metadata": {},
   "source": [
    "### Transcriptome"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4158acff",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i path_filtered -o adata_filtered\n",
    "suppressPackageStartupMessages(library(Seurat))\n",
    "suppressPackageStartupMessages(library('SingleCellExperiment', lib='/home/icb/felix.drost/miniconda3/envs/rds/lib/R/lib'))\n",
    "rds <- readRDS(path_filtered)\n",
    "adata_filtered <- as.SingleCellExperiment(rds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "e7c50128",
   "metadata": {},
   "outputs": [],
   "source": [
    "adata_filtered.obs['nFeature_RNA'] = adata_filtered.obs['nFeature_RNA'].values.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "5f3d9b3d",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig.ident' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'db.class' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'HPCA.first.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'HPCA.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'HPCA.pruned.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'DICE.first.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'DICE.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'DICE.pruned.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'functional.cluster' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'consensus.major' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'consensus.Tcell' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'barcode' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTgene' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTnt' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTaa' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTstrict' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'cloneType' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'SampleID' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'GEO_RNA' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Cohort' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Type' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Tissue' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Sample' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Sorted' as categorical\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 1593679 × 59326\n",
       "    obs: 'orig.ident', 'nCount_RNA', 'nFeature_RNA', 'mito.genes', 'db.weight.score', 'db.ratio', 'db.class', 'db.score', 'HPCA.first.labels', 'HPCA.labels', 'HPCA.pruned.labels', 'DICE.first.labels', 'DICE.labels', 'DICE.pruned.labels', 'functional.cluster', 'functional.cluster.conf', 'consensus.major', 'consensus.Tcell', 'barcode', 'CTgene', 'CTnt', 'CTaa', 'CTstrict', 'Frequency', 'cloneType', 'Treg_UCell', 'CD8_Activation_UCell', 'Antinflammatory_UCell', 'Anergy_UCell', 'Proinflammatory_UCell', 'Lipid_mediators_UCell', 'Glycolysis_UCell', 'TCA_cycle_UCell', 'PPS_UCell', 'Glycogen_Metabolism_UCell', 'Glucose_Deprivation_UCell', 'M1.Macro_UCell', 'M2.Macro_UCell', 'Cytolytic_UCell', 'T1_Interferon_UCell', 'T2_Interferon_UCell', 'Hypoxia_UCell', 'T_Cell_Terminal_Differentiation_UCell', 'G1S_UCell', 'G2M_UCell', 'SampleID', 'GEO_RNA', 'Cohort', 'Type', 'Tissue', 'Sample', 'Sorted', 'RNA_snn_res.0.8', 'RNA_snn_res.1.2', 'RNA_snn_res.1.6', 'seurat_clusters', 'ident'\n",
       "    obsm: 'X_pca', 'HARMONY', 'X_umap'\n",
       "    layers: 'logcounts'"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "adata_filtered.write_h5ad(out_filtered)\n",
    "adata_filtered"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ebfd2c5c",
   "metadata": {},
   "source": [
    "### Second experiment"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "9da084fc",
   "metadata": {},
   "outputs": [],
   "source": [
    "%%R -i path_gse123 -o gse123\n",
    "suppressPackageStartupMessages(library(Seurat))\n",
    "suppressPackageStartupMessages(library('SingleCellExperiment', lib='/home/icb/felix.drost/miniconda3/envs/rds/lib/R/lib'))\n",
    "rds <- readRDS(path_gse123)\n",
    "gse123 <- as.SingleCellExperiment(rds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "bf7d7bf7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 78034 × 23467\n",
       "    obs: 'orig.ident', 'nCount_RNA', 'nFeature_RNA', 'mito.genes', 'db.weight.score', 'db.ratio', 'db.class', 'db.score', 'SampleID', 'GEO_RNA', 'Cohort', 'Type', 'Tissue', 'Sample', 'Sorted', 'HPCA.first.labels', 'HPCA.labels', 'HPCA.pruned.labels', 'DICE.first.labels', 'DICE.labels', 'DICE.pruned.labels', 'functional.cluster', 'functional.cluster.conf', 'consensus.major', 'consensus.Tcell', 'Treg_UCell', 'CD8_Activation_UCell', 'Antinflammatory_UCell', 'Anergy_UCell', 'Proinflammatory_UCell', 'Lipid_mediators_UCell', 'Glycolysis_UCell', 'TCA_cycle_UCell', 'PPS_UCell', 'Glycogen_Metabolism_UCell', 'Glucose_Deprivation_UCell', 'M1.Macro_UCell', 'M2.Macro_UCell', 'Cytolytic_UCell', 'T1_Interferon_UCell', 'T2_Interferon_UCell', 'Hypoxia_UCell', 'T_Cell_Terminal_Differentiation_UCell', 'G1S_UCell', 'G2M_UCell', 'barcode', 'CTgene', 'CTnt', 'CTaa', 'CTstrict', 'Frequency', 'cloneType', 'ident'\n",
       "    layers: 'logcounts'"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse123"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "49df864f",
   "metadata": {},
   "outputs": [],
   "source": [
    "gse123.obs['nFeature_RNA'] = gse123.obs['nFeature_RNA'].values.astype(int)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "74cf91be",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'orig.ident' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'db.class' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'SampleID' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'GEO_RNA' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Cohort' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Type' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Tissue' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Sample' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'Sorted' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'HPCA.first.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'HPCA.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'HPCA.pruned.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'DICE.first.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'DICE.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'DICE.pruned.labels' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'functional.cluster' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'consensus.major' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'consensus.Tcell' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'barcode' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTgene' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTnt' as categorical\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTaa' as categorical\n",
      "/home/icb/felix.drost/miniconda3/envs/rds/lib/python3.8/site-packages/anndata/_core/anndata.py:1228: FutureWarning: The `inplace` parameter in pandas.Categorical.reorder_categories is deprecated and will be removed in a future version. Reordering categories will always return a new Categorical object.\n",
      "  c.reorder_categories(natsorted(c.categories), inplace=True)\n",
      "... storing 'CTstrict' as categorical\n"
     ]
    },
    {
     "data": {
      "text/plain": [
       "AnnData object with n_obs × n_vars = 78034 × 23467\n",
       "    obs: 'orig.ident', 'nCount_RNA', 'nFeature_RNA', 'mito.genes', 'db.weight.score', 'db.ratio', 'db.class', 'db.score', 'SampleID', 'GEO_RNA', 'Cohort', 'Type', 'Tissue', 'Sample', 'Sorted', 'HPCA.first.labels', 'HPCA.labels', 'HPCA.pruned.labels', 'DICE.first.labels', 'DICE.labels', 'DICE.pruned.labels', 'functional.cluster', 'functional.cluster.conf', 'consensus.major', 'consensus.Tcell', 'Treg_UCell', 'CD8_Activation_UCell', 'Antinflammatory_UCell', 'Anergy_UCell', 'Proinflammatory_UCell', 'Lipid_mediators_UCell', 'Glycolysis_UCell', 'TCA_cycle_UCell', 'PPS_UCell', 'Glycogen_Metabolism_UCell', 'Glucose_Deprivation_UCell', 'M1.Macro_UCell', 'M2.Macro_UCell', 'Cytolytic_UCell', 'T1_Interferon_UCell', 'T2_Interferon_UCell', 'Hypoxia_UCell', 'T_Cell_Terminal_Differentiation_UCell', 'G1S_UCell', 'G2M_UCell', 'barcode', 'CTgene', 'CTnt', 'CTaa', 'CTstrict', 'Frequency', 'cloneType', 'ident'\n",
       "    layers: 'logcounts'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse123.write_h5ad(out_gse123)\n",
    "gse123"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "95e1997f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>barcode</th>\n",
       "      <th>is_cell</th>\n",
       "      <th>contig_id</th>\n",
       "      <th>high_confidence</th>\n",
       "      <th>length</th>\n",
       "      <th>chain</th>\n",
       "      <th>v_gene</th>\n",
       "      <th>d_gene</th>\n",
       "      <th>j_gene</th>\n",
       "      <th>c_gene</th>\n",
       "      <th>full_length</th>\n",
       "      <th>productive</th>\n",
       "      <th>cdr3</th>\n",
       "      <th>cdr3_nt</th>\n",
       "      <th>reads</th>\n",
       "      <th>umis</th>\n",
       "      <th>raw_clonotype_id</th>\n",
       "      <th>raw_consensus_id</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>AAACCTGAGAAGGTTT-NC4</td>\n",
       "      <td>True</td>\n",
       "      <td>AAACCTGAGAAGGTTT-NC4_contig_1</td>\n",
       "      <td>True</td>\n",
       "      <td>459</td>\n",
       "      <td>TRA</td>\n",
       "      <td>TRAV13-1</td>\n",
       "      <td>None</td>\n",
       "      <td>TRAJ9</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CAARDTGGFKTIF</td>\n",
       "      <td>TGTGCAGCAAGAGATACTGGAGGCTTCAAAACTATCTTT</td>\n",
       "      <td>4400</td>\n",
       "      <td>5</td>\n",
       "      <td>clonotype48</td>\n",
       "      <td>clonotype48_consensus_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>AAACCTGAGAAGGTTT-NC4</td>\n",
       "      <td>True</td>\n",
       "      <td>AAACCTGAGAAGGTTT-NC4_contig_2</td>\n",
       "      <td>True</td>\n",
       "      <td>498</td>\n",
       "      <td>TRB</td>\n",
       "      <td>TRBV20-1</td>\n",
       "      <td>None</td>\n",
       "      <td>TRBJ2-1</td>\n",
       "      <td>TRBC2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CSAGSGGLIHEQFF</td>\n",
       "      <td>TGCAGTGCTGGAAGCGGGGGGTTGATTCATGAGCAGTTCTTC</td>\n",
       "      <td>7972</td>\n",
       "      <td>14</td>\n",
       "      <td>clonotype48</td>\n",
       "      <td>clonotype48_consensus_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>AAACCTGAGACAGGCT-CT7</td>\n",
       "      <td>True</td>\n",
       "      <td>AAACCTGAGACAGGCT-CT7_contig_1</td>\n",
       "      <td>True</td>\n",
       "      <td>475</td>\n",
       "      <td>TRB</td>\n",
       "      <td>TRBV4-1</td>\n",
       "      <td>None</td>\n",
       "      <td>TRBJ1-2</td>\n",
       "      <td>TRBC1</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CASSQVWGKKSTF</td>\n",
       "      <td>TGCGCCAGCAGCCAAGTATGGGGGAAGAAGAGTACCTTC</td>\n",
       "      <td>10886</td>\n",
       "      <td>30</td>\n",
       "      <td>clonotype12</td>\n",
       "      <td>clonotype12_consensus_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>AAACCTGAGACAGGCT-CT7</td>\n",
       "      <td>True</td>\n",
       "      <td>AAACCTGAGACAGGCT-CT7_contig_2</td>\n",
       "      <td>True</td>\n",
       "      <td>631</td>\n",
       "      <td>TRA</td>\n",
       "      <td>TRAV41</td>\n",
       "      <td>None</td>\n",
       "      <td>TRAJ52</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CAVRFYLAGGTSYGKLTF</td>\n",
       "      <td>TGTGCTGTCAGATTCTACCTTGCTGGTGGTACTAGCTATGGAAAGC...</td>\n",
       "      <td>5056</td>\n",
       "      <td>12</td>\n",
       "      <td>clonotype12</td>\n",
       "      <td>clonotype12_consensus_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>AAACCTGAGACAGGCT-CT7</td>\n",
       "      <td>True</td>\n",
       "      <td>AAACCTGAGACAGGCT-CT7_contig_3</td>\n",
       "      <td>True</td>\n",
       "      <td>344</td>\n",
       "      <td>TRA</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>TRAJ45</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>False</td>\n",
       "      <td>False</td>\n",
       "      <td>None</td>\n",
       "      <td>None</td>\n",
       "      <td>504</td>\n",
       "      <td>2</td>\n",
       "      <td>clonotype12</td>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200497</th>\n",
       "      <td>TTTGTCATCTTGGGTA-NC6</td>\n",
       "      <td>True</td>\n",
       "      <td>TTTGTCATCTTGGGTA-NC6_contig_1</td>\n",
       "      <td>True</td>\n",
       "      <td>574</td>\n",
       "      <td>TRA</td>\n",
       "      <td>TRAV17</td>\n",
       "      <td>None</td>\n",
       "      <td>TRAJ8</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CSTGFQKLVF</td>\n",
       "      <td>TGTTCTACAGGCTTTCAGAAACTTGTATTT</td>\n",
       "      <td>932</td>\n",
       "      <td>4</td>\n",
       "      <td>clonotype1744</td>\n",
       "      <td>clonotype1744_consensus_3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200498</th>\n",
       "      <td>TTTGTCATCTTGGGTA-NC6</td>\n",
       "      <td>True</td>\n",
       "      <td>TTTGTCATCTTGGGTA-NC6_contig_2</td>\n",
       "      <td>True</td>\n",
       "      <td>547</td>\n",
       "      <td>TRA</td>\n",
       "      <td>TRAV12-2</td>\n",
       "      <td>None</td>\n",
       "      <td>TRAJ22</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CAVNHPGSARQLTF</td>\n",
       "      <td>TGTGCCGTGAACCACCCTGGTTCTGCAAGGCAACTGACCTTT</td>\n",
       "      <td>1174</td>\n",
       "      <td>4</td>\n",
       "      <td>clonotype1744</td>\n",
       "      <td>clonotype1744_consensus_1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200499</th>\n",
       "      <td>TTTGTCATCTTGGGTA-NC6</td>\n",
       "      <td>True</td>\n",
       "      <td>TTTGTCATCTTGGGTA-NC6_contig_3</td>\n",
       "      <td>True</td>\n",
       "      <td>525</td>\n",
       "      <td>TRB</td>\n",
       "      <td>TRBV13</td>\n",
       "      <td>None</td>\n",
       "      <td>TRBJ2-5</td>\n",
       "      <td>TRBC2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CASSDIQGRAETQYF</td>\n",
       "      <td>TGTGCCAGCAGCGACATACAGGGTAGGGCAGAGACCCAGTACTTC</td>\n",
       "      <td>3744</td>\n",
       "      <td>18</td>\n",
       "      <td>clonotype1744</td>\n",
       "      <td>clonotype1744_consensus_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200500</th>\n",
       "      <td>TTTGTCATCTTGTTTG-NC3</td>\n",
       "      <td>True</td>\n",
       "      <td>TTTGTCATCTTGTTTG-NC3_contig_1</td>\n",
       "      <td>True</td>\n",
       "      <td>495</td>\n",
       "      <td>TRA</td>\n",
       "      <td>TRAV9-2</td>\n",
       "      <td>None</td>\n",
       "      <td>TRAJ13</td>\n",
       "      <td>TRAC</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CALIPAAGYQKVTF</td>\n",
       "      <td>TGTGCTCTGATTCCTGCCGCGGGTTACCAGAAAGTTACCTTT</td>\n",
       "      <td>13602</td>\n",
       "      <td>17</td>\n",
       "      <td>clonotype52013</td>\n",
       "      <td>clonotype52013_consensus_2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>200501</th>\n",
       "      <td>TTTGTCATCTTGTTTG-NC3</td>\n",
       "      <td>True</td>\n",
       "      <td>TTTGTCATCTTGTTTG-NC3_contig_2</td>\n",
       "      <td>True</td>\n",
       "      <td>507</td>\n",
       "      <td>TRB</td>\n",
       "      <td>TRBV12-3</td>\n",
       "      <td>None</td>\n",
       "      <td>TRBJ2-6</td>\n",
       "      <td>TRBC2</td>\n",
       "      <td>True</td>\n",
       "      <td>True</td>\n",
       "      <td>CASSDSSSGANVLTF</td>\n",
       "      <td>TGTGCCAGCAGTGATAGTTCCTCTGGGGCCAACGTCCTGACTTTC</td>\n",
       "      <td>21614</td>\n",
       "      <td>33</td>\n",
       "      <td>clonotype52013</td>\n",
       "      <td>clonotype52013_consensus_1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>200501 rows × 18 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                     barcode  is_cell                      contig_id  \\\n",
       "1       AAACCTGAGAAGGTTT-NC4     True  AAACCTGAGAAGGTTT-NC4_contig_1   \n",
       "2       AAACCTGAGAAGGTTT-NC4     True  AAACCTGAGAAGGTTT-NC4_contig_2   \n",
       "3       AAACCTGAGACAGGCT-CT7     True  AAACCTGAGACAGGCT-CT7_contig_1   \n",
       "4       AAACCTGAGACAGGCT-CT7     True  AAACCTGAGACAGGCT-CT7_contig_2   \n",
       "5       AAACCTGAGACAGGCT-CT7     True  AAACCTGAGACAGGCT-CT7_contig_3   \n",
       "...                      ...      ...                            ...   \n",
       "200497  TTTGTCATCTTGGGTA-NC6     True  TTTGTCATCTTGGGTA-NC6_contig_1   \n",
       "200498  TTTGTCATCTTGGGTA-NC6     True  TTTGTCATCTTGGGTA-NC6_contig_2   \n",
       "200499  TTTGTCATCTTGGGTA-NC6     True  TTTGTCATCTTGGGTA-NC6_contig_3   \n",
       "200500  TTTGTCATCTTGTTTG-NC3     True  TTTGTCATCTTGTTTG-NC3_contig_1   \n",
       "200501  TTTGTCATCTTGTTTG-NC3     True  TTTGTCATCTTGTTTG-NC3_contig_2   \n",
       "\n",
       "        high_confidence  length chain    v_gene d_gene   j_gene c_gene  \\\n",
       "1                  True     459   TRA  TRAV13-1   None    TRAJ9   TRAC   \n",
       "2                  True     498   TRB  TRBV20-1   None  TRBJ2-1  TRBC2   \n",
       "3                  True     475   TRB   TRBV4-1   None  TRBJ1-2  TRBC1   \n",
       "4                  True     631   TRA    TRAV41   None   TRAJ52   TRAC   \n",
       "5                  True     344   TRA      None   None   TRAJ45   TRAC   \n",
       "...                 ...     ...   ...       ...    ...      ...    ...   \n",
       "200497             True     574   TRA    TRAV17   None    TRAJ8   TRAC   \n",
       "200498             True     547   TRA  TRAV12-2   None   TRAJ22   TRAC   \n",
       "200499             True     525   TRB    TRBV13   None  TRBJ2-5  TRBC2   \n",
       "200500             True     495   TRA   TRAV9-2   None   TRAJ13   TRAC   \n",
       "200501             True     507   TRB  TRBV12-3   None  TRBJ2-6  TRBC2   \n",
       "\n",
       "        full_length  productive                cdr3  \\\n",
       "1              True        True       CAARDTGGFKTIF   \n",
       "2              True        True      CSAGSGGLIHEQFF   \n",
       "3              True        True       CASSQVWGKKSTF   \n",
       "4              True        True  CAVRFYLAGGTSYGKLTF   \n",
       "5             False       False                None   \n",
       "...             ...         ...                 ...   \n",
       "200497         True        True          CSTGFQKLVF   \n",
       "200498         True        True      CAVNHPGSARQLTF   \n",
       "200499         True        True     CASSDIQGRAETQYF   \n",
       "200500         True        True      CALIPAAGYQKVTF   \n",
       "200501         True        True     CASSDSSSGANVLTF   \n",
       "\n",
       "                                                  cdr3_nt  reads  umis  \\\n",
       "1                 TGTGCAGCAAGAGATACTGGAGGCTTCAAAACTATCTTT   4400     5   \n",
       "2              TGCAGTGCTGGAAGCGGGGGGTTGATTCATGAGCAGTTCTTC   7972    14   \n",
       "3                 TGCGCCAGCAGCCAAGTATGGGGGAAGAAGAGTACCTTC  10886    30   \n",
       "4       TGTGCTGTCAGATTCTACCTTGCTGGTGGTACTAGCTATGGAAAGC...   5056    12   \n",
       "5                                                    None    504     2   \n",
       "...                                                   ...    ...   ...   \n",
       "200497                     TGTTCTACAGGCTTTCAGAAACTTGTATTT    932     4   \n",
       "200498         TGTGCCGTGAACCACCCTGGTTCTGCAAGGCAACTGACCTTT   1174     4   \n",
       "200499      TGTGCCAGCAGCGACATACAGGGTAGGGCAGAGACCCAGTACTTC   3744    18   \n",
       "200500         TGTGCTCTGATTCCTGCCGCGGGTTACCAGAAAGTTACCTTT  13602    17   \n",
       "200501      TGTGCCAGCAGTGATAGTTCCTCTGGGGCCAACGTCCTGACTTTC  21614    33   \n",
       "\n",
       "       raw_clonotype_id            raw_consensus_id  \n",
       "1           clonotype48     clonotype48_consensus_1  \n",
       "2           clonotype48     clonotype48_consensus_2  \n",
       "3           clonotype12     clonotype12_consensus_2  \n",
       "4           clonotype12     clonotype12_consensus_1  \n",
       "5           clonotype12                        None  \n",
       "...                 ...                         ...  \n",
       "200497    clonotype1744   clonotype1744_consensus_3  \n",
       "200498    clonotype1744   clonotype1744_consensus_1  \n",
       "200499    clonotype1744   clonotype1744_consensus_2  \n",
       "200500   clonotype52013  clonotype52013_consensus_2  \n",
       "200501   clonotype52013  clonotype52013_consensus_1  \n",
       "\n",
       "[200501 rows x 18 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "gse144 = pd.read_csv(path_gse144, index_col=0)\n",
    "gse144.to_csv(out_gse144)\n",
    "gse144"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c446bd98",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python [conda env:rds] *",
   "language": "python",
   "name": "conda-env-rds-py"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
